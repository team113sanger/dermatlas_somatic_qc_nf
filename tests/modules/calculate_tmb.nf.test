nextflow_process {

    name "Test Process CALCULATE_SAMPLE_TMB"
    script "modules/calculate_tmb.nf"
    process "CALCULATE_SAMPLE_TMB"

    test("Should run without failures") {

        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
            process {
                """
                input[0] = tuple([empty: 'test'], file("$baseDir/tests/testdata/example_maf.maf"))
                input[1] = 48.225157
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
