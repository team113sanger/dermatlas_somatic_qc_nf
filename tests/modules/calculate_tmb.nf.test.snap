{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "mutations_per_Mb.tsv:md5,933d7a1a7a8a90a79f9bf3116338e237"
                ],
                "tmb": [
                    "mutations_per_Mb.tsv:md5,933d7a1a7a8a90a79f9bf3116338e237"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-07-26T19:33:36.28603"
    }
}